microsoft/PowerToys
# Fix Settings crash on clearing FancyZones Hex Input

## Summary of the Pull Request

_What is this about?_
This PR fixes a crash that occurs in Settings on clicking the clear button in the WinUI ColorPicker element in FZ Settings.

## PR Checklist
* [X] Applies to #6022 

## Info on Pull Request

_What does this include?_
- Since there was no exception trace for this crash (the only available information was the exit code was 0xc000041d, which is not documented) I checked with the [Xaml Island sample](https://github.com/microsoft/Xaml-Islands-Samples) by adding a ColorPicker element to validate if it was a Xaml Island bug, but I could not repro the error. After deleting several code files/updating nuget packages to identify the differences between the two projects I found that declaring the ColorPicker element without the `muxc:` prefix fixes the issue. The `muxc` namespace is not required because of this code in the App.xaml file:
https://github.com/microsoft/PowerToys/blob/0349383d082e604a6111e79f1253645ae7bb0473/src/core/Microsoft.PowerToys.Settings.UI/App.xaml#L10
- I was able to repro this on the [Xaml Control Gallery repo](https://github.com/microsoft/Xaml-Controls-Gallery) as well, so it is most likely a WinUI bug. An issue has been created on their repo and it is linked in the above issue. The current changes are a workaround until the bug is fixed in the WinUI nuget package.

## Validation Steps Performed

_How does someone test & validate?_
- Launch PowerToys Settings and open the FancyZones page.
- Try to clear all the input boxes in the ColorPicker drop downs to ensure it does not crash on any of them.
